pkg/multiclustermanager/app.go

```diff
@@ -227,5 +225,14 @@ func (m *mcm) Start(ctx context.Context) error {
 		return nil
 	})
 
+	// Keep KDM handlers out of the above OnLeader block to be executed on all Pod replicas
+	err = m.wranglerContext.StartWithTransaction(ctx, func(ctx context.Context) error {
```

Comment:
Can we start the same `wranglerContext` twice? I think the answer to it is no. That could be why you see the  "no version info found in KDM"  in UI when you create a custom cluster during testing.  ( Hi @kinarashah , do you know the answer to that question or have more insights to share? ) 

Also, we need to register the handler with the `ScaledContext` to let the controller run on all rancher replicas.